File: Program's_Contributed_By_Contributors/JavaScript_Programs/quick-sort.js
const quickSort = (arr) => {
    if (arr.length <= 1) {
      return arr;
    }
  
    let pivot = arr[0];
    let leftArr = [];
    let rightArr = [];
  
    for (let i = 1; i < arr.length; i++) {
      if (arr[i] < pivot) {
        leftArr.push(arr[i]);
      } else {
        rightArr.push(arr[i]);
      }
    }
  
    return [...quickSort(leftArr), pivot, ...quickSort(rightArr)];
  };

let myArray = [3, 7, 2, 5, 1, 4, 6, 8];
console.log(quicksort(myArray)); // Output: [1,2,3,4,5,6,7,8]

File: Program's_Contributed_By_Contributors/C++ Programs/Array/quicksort.cpp
{
    public:
    //Function to sort an array using quick sort algorithm.
    void quickSort(int arr[], int low, int high)
    {
        // code here
         if(low>high) return;
         int p=  partition(arr,low,high);
         quickSort(arr,low,p-1);
         quickSort(arr,p+1,high);
    }
    
    public:
    int partition (int arr[], int low, int high)
    {
        int pivot=arr[high];
        int i=low,j=low-1;
        
        while(i<high)
        {
            if(arr[i]<=pivot)
            {
                swap(arr[i],arr[j]);
                j++;
            }
            i++;
        }
        
        swap(arr[j],arr[high]);
        return j;
       
    }
};
